### PR TITLE
Port mainapp caelum rc4

### PR DESCRIPTION
**Pull request by NathanaelGandhi**

### CODE REVIEW

File: fsw/src/ci_custom.c [hunk 1]
```diff
@@ -93,84 +91,73 @@ static CI_CustomData_t g_CI_CustomData;
...
 {
-    int32 iStatus = CI_ERROR;
-    uint32 taskId = 0;
+    int32               iStatus = CI_ERROR;
+    CFE_ES_TaskId_t    *taskId  = NULL;
     IO_TransUdpConfig_t config;
 
     /* Set Config parameters */
-    CFE_PSP_MemSet((void *) &config, 0x0, sizeof(IO_TransUdpConfig_t));
+    CFE_PSP_MemSet((void *)&config, 0x0, sizeof(IO_TransUdpConfig_t));
```
Comment (LucienMorey) [inline fsw/src/ci_custom.c:99]: memset comment

File: fsw/src/ci_custom.c [hunk 2]
```diff
@@ -93,84 +91,73 @@ static CI_CustomData_t g_CI_CustomData;
...
 /******************************************************************************/
 /** \brief Custom response to CI_ENABLE_TO_CC cmd code
 *******************************************************************************/
-void CI_CustomEnableTO(CFE_SB_MsgPtr_t cmdMsgPtr)
+void CI_CustomEnableTO(const CFE_SB_Buffer_t *cmdMsgPtr)
 {
     /* NOTE: In this case, we are simply piping the cmd to TO. */
-    CFE_PSP_MemCpy((void *) &g_CI_CustomData.toEnableCmd, 
-                   (void *) cmdMsgPtr, sizeof(TO_EnableOutputCmd_t));
+    CFE_PSP_MemCpy((void *)&g_CI_CustomData.toEnableCmd, (void *)cmdMsgPtr, sizeof(TO_EnableOutputCmd_t));
```
Comment (LucienMorey) [inline fsw/src/ci_custom.c:149]: why psp instead? this is the same as the other issue

File: fsw/src/ci_custom.c [hunk 3]
```diff
@@ -93,84 +91,73 @@ static CI_CustomData_t g_CI_CustomData;
...
     /* Setup the toEnableCmd */
-    CFE_SB_InitMsg((CFE_SB_MsgPtr_t) &g_CI_CustomData.toEnableCmd, 
-                   TO_APP_CMD_MID, sizeof(TO_EnableOutputCmd_t), FALSE); 
-    CFE_SB_SetCmdCode((CFE_SB_MsgPtr_t) &g_CI_CustomData.toEnableCmd, 
-                      TO_ENABLE_OUTPUT_CC);
-    CFE_SB_GenerateChecksum((CFE_SB_MsgPtr_t) &g_CI_CustomData.toEnableCmd);
-
-    /* Send the TO Enable Telemetry Output Message */    
-    CFE_SB_SendMsg((CFE_SB_MsgPtr_t) &g_CI_CustomData.toEnableCmd);
+    CFE_MSG_Init(&g_CI_CustomData.toEnableCmd.CmdHeader.Msg, TO_APP_CMD_MID, sizeof(TO_EnableOutputCmd_t));
```
Comment (LucienMorey) [inline fsw/src/ci_custom.c:152]:
```suggestion
    CFE_MSG_Init(&g_CI_CustomData.toEnableCmd.CmdHeader.Msg, CFE_SB_ValueToMsgId(TO_APP_CMD_MID), sizeof(TO_EnableOutputCmd_t));
```

**Review by LucienMorey — CHANGES_REQUESTED** [review]

File: fsw/src/ci_app.c [hunk 4]
```diff
@@ -178,157 +163,137 @@ int32 CI_AppInit(void)
...
-                   sizeof(g_CI_AppData.cSchPipeName));
-    strncpy(g_CI_AppData.cSchPipeName, "CI_SCH_PIPE", OS_MAX_API_NAME-1);
+    CFE_PSP_MemSet((void *)g_CI_AppData.cSchPipeName, '\0', sizeof(g_CI_AppData.cSchPipeName));
+    strncpy(g_CI_AppData.cSchPipeName, "CI_SCH_PIPE", OS_MAX_API_NAME - 1);
 
     /* Subscribe to Wakeup messages */
-    iStatus = CFE_SB_CreatePipe(&g_CI_AppData.SchPipeId,
-                                 g_CI_AppData.usSchPipeDepth,
-                                 g_CI_AppData.cSchPipeName);
+    iStatus = CFE_SB_CreatePipe(&g_CI_AppData.SchPipeId, g_CI_AppData.usSchPipeDepth, g_CI_AppData.cSchPipeName);
```
[the remark this answers is not in the document]
Comment (NathanaelGandhi) [reply fsw/src/ci_app.c:206]: CFE_SB_PipeId_t CmdPipeId;                     /**< Command Pipe ID */
ci_app.h

File: fsw/src/ci_app.c [hunk 5]
```diff
@@ -178,157 +163,137 @@ int32 CI_AppInit(void)
...
+
 /*****************************************************************************/
 /** \brief Initialize the Event Filter Table.
 ******************************************************************************/
 int32 CI_InitEvent(void)
 {
-    int32  iStatus=CFE_SUCCESS;
-    int32  ii = 0;
+    int32 iStatus = CFE_SUCCESS;
+    int32 ii      = 0;
```
[the remark this answers is not in the document]
Comment (NathanaelGandhi) [reply fsw/src/ci_app.c:173]: Just a counter for all the event filter table items

File: fsw/platform_inc/ci_platform_cfg.h [hunk 6]
```diff
@@ -32,47 +33,46 @@ extern "C" {
...
+#define CI_SCH_PIPE_DEPTH 10
+#define CI_CMD_PIPE_DEPTH 10
+#define CI_TLM_PIPE_DEPTH 10
 
-
-#define CI_CUSTOM_UDP_PORT 5010
-#define CI_CUSTOM_UDP_ADDR IO_TRANS_UDP_INADDR_ANY
-#define CI_CUSTOM_UDP_TIMEOUT 100
-#define CI_CUSTOM_MAX_IP_STRING_SIZE  16 
+#define CI_CUSTOM_UDP_PORT           5010
```
[the remark this answers is not in the document]
Comment (NathanaelGandhi) [reply fsw/platform_inc/ci_platform_cfg.h:42]: Making a note to change our port on openc3

**Pull request merged**